### ISSUE TITLE
Using web components in a Mesop app requires a lot of configuration

### ISSUE THREAD

**Issue by richard-to** [google/mesop#678]
When using web components in a Mesop app, we need to make sure to add the appropriate stylesheets and CSP security configurations each time for each app.

It could be nice if the web components could be configured with some of the CSP configurations required to get it to work. Then we can merge into the page security policy.

Stylesheet would be good too, though I guess there are cases where one may want to use a different style sheet like in the markedjs web component where we load a stylesheet for the code highlighting and there are different themes to choose from.

**Comment by wwwillchen** [comment]
Yeah, I think the usability of web components is definitely not great right now. The one concern I have about this is that you may depend on a web component very loose settings (e.g. disabling trusted types, and very permissive allow-list) and it may not be obvious that your app is running very loose security settings. The alternative approach I have is to make it very easy to auto-update the security policy based on CSP violations so then, it's 1) explicit and 2) relatively straightforward to fix CSP errors. WDYT?

**Comment by richard-to** [comment]
That's a great point about the security settings not being obvious. I think your suggestion sounds good. Seems like it would resolve some other general CSP violation error handling issues as well.

Curious to hear more details about it, especially 2).
